### DISCUSSION TITLE
Error using impute

### DISCUSSION THREAD

**Discussion by ghost** [blue-yonder/tsfresh#1022] [General]
Sorry for this silly issue I'm having but i'm fairly new to tsfresh.

Ultimately, I'm having issues imputing. I am extracting features from my data set, but when I try to impute I get the following error: "AttributeError: 'list' object has no attribute 'values'"

I get this error if I run the example feature extraction using the robot arm data or with my own data.

My data:

      id     time  values
0      1  0.00025 -0.0333
1      1  0.00050 -0.0350
2      1  0.00075 -0.0350
3      1  0.00100 -0.0300
4      1  0.00125 -0.0300
...   ..      ...     ...
4995   1  1.24900 -0.1117
4996   1  1.24925 -0.1050
4997   1  1.24950 -0.1050
4998   1  1.24975 -0.0933
4999   1  1.25000 -0.0933

Then I run:
extracted_features = tsfresh.extract_features(mydata, column_id="id", column_sort="time", pivot=False)

Which gives me:
[((1,), 'values__variance_larger_than_standard_deviation', False), ((1,), 'values__has_duplicate_max', False), ((1,), 'values__has_duplicate_min', False), ((1,), 'values__has_duplicate', True), ((1,), 'values__sum_values', -8.709800000000001), ((1,), 'values__abs_energy', 21.485108959999987), ((1,), 'values__mean_abs_change', 0.012017883576715344), ((1,), 'values__mean_change', -1.2002400480096017e-05), ((1,), 'values__mean_second_derivative_central', 1.7006802721088437e-07), ((1,), 'values__median', -0.0033), ((1,), 'values__mean', -0.0017419600000000003), ((1,), 'values__length', 5000), ((1,), 'values__standard_deviation', 0.06552852331129094), ((1,), 'values__variation_coefficient', -37.617696911117896), ....

etc

and if I type:
impute(extracted_features)

I get an error:
AttributeError: 'list' object has no attribute 'values'

I'm sure I'm doing something very silly but I haven't been able to figure it out. I've also tried converting my list back into a dataframe but then I get a different error related to isfinite. Seems like everyone has just been imputing directly following their feature extraction so I can't figure out my problem

**Comment by nils-braun** [comment]
Hi @FyLapS !
The problem might be that the impute function (at least if you are using one from tsfresh), is expecting the data to be in the "normal" pivoted dataframe version. Is there a reason why you set `pivot=False` (which is not the default)?

**Reply by ghost** [reply]
Thanks very much for writing! Do you know how to hand this with the pivot, or example is there something I can do to get it in the right form?
Also, if I keep the pivot I get another error (if all I change in my code is take out "pivot=False"):
TypeError: Setting a MultiIndex dtype to anything other than object is not supported

**Reply by nils-braun** [reply]
Yes, you want to remove the `pivot=False` from the call.
To solve the other error: this is currently being caused by newer versions of pandas.
Either make sure your ID column is of type object, like
```python
mydata["id"] = mydata["id"].astype(object)
```
before the feature extraction or try with an older version of pandas (below 2.0.0).

We are working on a fix for this!

**Reply by ghost** [reply]
Thanks, Nils! I didn't write earlier, but wanted to let you know it works now. Thanks very much for helping!!
